release notes/v0.40.0.md

```diff
@@ -0,0 +1,253 @@
+      clearTimeout(timeout1id);
+      clearTimeout(timeout2id);
+      console.log(`VU ${__VU}:${id}: disconnected`);
+    })
+  });
+}
+```
+
+It is important to note that no k6 iterations do not finish if there is any WebSocket still open or if a timeout/interval is not cleared or triggered. This means that you need to be careful when writing the script to clear all interval and close the websocket in some point in time. k6 will still kill the whole process if it takes too long to stop after the maximum test duration is reached.
+
+Current issues and future improvements for the WebSockets API can be found in its [issue tracker](https://github.com/grafana/xk6-websockets/issues). Currently, documentation is available through [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket) although some features are not yet supported:
+- no Blob binary type - ArrayBuffer is the default
+- no `onMessage` and co. - only addEventListener
+
+## First class support for JavaScript Classes
+
+As part of [updating goja](https://github.com/grafana/k6/pull/2610) k6 got native, as in not by transpiling by the internal babel, support for classes 🎉. Because this actually implements classes as they described in the latest ECMAScript specification this also means we get a ton of additional class features that were never previously supported such as private fields. Additionally at least one bug [#1763](https://github.com/grafana/k6/issues/1763) was fixed as a result of this, but probably many more as well.
+
+Due to this fairly big change there are chances that some code will now behave differently. Please report any issues, though take into consideration that it's possible that the new behavior is just the correct one.
+
+Other updates from goja are:
+- optimizations around using strings and some access patterns
+- support for `\u{01234}` Unicode point syntax in regexp
+- Fixed a case where interrupting the VM would not have worked, especially around try/catch usage. This was particularly problematic for k6 as this could lead to the whole k6 hanging forever. [#2600](https://github.com/grafana/k6/pull/2600)
+
+Many thanks to @dop251 for continuing to improve goja! 
+
+### New Test runtime for module extension developers [#2598](https://github.com/grafana/k6/issues/2598)
+
+While we are developing extensions internally it constantly comes up that you need to do create the same structure over and over again when you run stuff in tests. With the addition of the event loops it is now required to setup it as well. Even k6 team members get parts of this wrong every once in a while, so we decided to add a small type to be used by (extension) module developers to write tests easier. [#2602](https://github.com/grafana/k6/pull/2602)
+
+[This API](https://pkg.go.dev/go.k6.io/k6@v0.39.1-0.20220819115243-e09bb8727786/js/modulestest#Runtime) will likely change and evolve as we add more functionality or as we change the k6 internal API.
+
+
+# Breaking changes
```

Comment:
What if we combine this section with the first two breaking changes from the above? I believe you want to stress their importance, but on the other side, having two breaking-changes sections can confuse a bit, WDYT?

Reply:
Hmmm - I do repeat them here as well ;). 

But yeah, maybe we should ahve the "Breaking changes" on the top :thinking: Even the one that have paragraphs are fairly short.

This also removes the need to figure out all the ⚠ things 
WDYT @codebien

Reply:
So do we want not to mention it in the specific module's section but we will have an item in this list, am I understanding correct?

Reply:
The idea is to 
1. move the breaking changes section to the top
2. Move the two sections with breaking changes from the "New features and notable changes" to *just* the `Breaking changes` section

So I guess we agree on this - I will do it in a minute :)

Reply:
[b358c6a](https://github.com/grafana/k6/pull/2669/commits/b358c6aa17f7f9e5eb613996f5516113422517ac)